lib/core/utils/cache_interceptor.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:project_kepler/core/global.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'dart:convert';
+
+class CacheInterceptor extends Interceptor {
+  final SharedPreferences sharedPreferences;
+
+  CacheInterceptor(this.sharedPreferences);
+
+  @override
+  void onRequest(
+      RequestOptions options, RequestInterceptorHandler handler) async {
+    final key = options.uri.toString();
+    final cachedData = sharedPreferences.getString(key);
+
+    if (cachedData != null) {
+      final data = jsonDecode(cachedData);
+
+      final expiryDate = DateTime.tryParse(data['expiryDate']);
+
+      logger.d(
+          "Cache expiry in  ${expiryDate?.difference(DateTime.now()).inSeconds} seconds");
+      if (expiryDate != null && DateTime.now().isBefore(expiryDate)) {
+        // If cached data is not expired, return it directly
+        logger.d('Returning cached data');
+        return handler.resolve(Response(
+          requestOptions: options,
+          statusCode: 200,
+          data: data['data'],
+        ));
+      }
+    }
+    logger.d('No valid cached data, proceed with request');
+    // No valid cached data, proceed with request
+    handler.next(options);
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) async {
+    // Cache duration from .env file
+    final cacheDuration = dotenv.env['CACHE_DURATION'];
+
+    final duration = cacheDuration != null
+        ? Duration(seconds: int.parse(cacheDuration))
```

Comment:
Using `int.parse` without try-catch for parsing `cacheDuration` can cause an unhandled exception if the environment variable is not a valid integer.

```diff
-        ? Duration(seconds: int.parse(cacheDuration))
+        ? Duration(seconds: int.tryParse(cacheDuration) ?? 60)
```